src/sentry/models/sentryapp.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import absolute_import
+
+import six
+import uuid
+
+from bitfield import BitField
+from django.db import models
+from django.utils import timezone
+
+from sentry.db.models import (ArrayField, FlexibleForeignKey, ParanoidModel)
+from sentry.models import ApiScopes
+from sentry.utils.strings import dasherize
+
+
+class SentryApp(ParanoidModel):
+    __core__ = True
+
+    application = models.OneToOneField('sentry.ApiApplication',
```

Comment:
These OneToOneFields are where you'll probably hit an issue due to the soft deletions, and I don't see where this is being handled.

I guess would an existing `SentryApp` row just be undeleted if it were re-added? Instead of adding a new row?

Reply:
Yea, I think we'd just want to "undelete" it. That way history is left intact.